### PR TITLE
Made some changes to ComicInfo metadata

### PR DESCRIPTION
**Pull request by Shamicen**
I made some changes to `ComicInfo` metadata generation:

The `web field` had a schema change and can now contain multiple space separated reference website links. Therefore ComicInfo metadata will from now on include the `source URL` and all available `tracker URLs`.
For more information look at this issue https://github.com/anansi-project/comicinfo/issues/56 and this PR https://github.com/anansi-project/comicinfo/pull/57

I changed the `translator field` to use the source name if no `scanlator` name is available.

### CODE REVIEW

**Comment by AntsyLich** [comment]
> I changed the `translator field` to use the source name if no `scanlator` name is available.

Not sure i like that change

**Comment by Shamicen** [comment]
I was planning to make it so that `LocalSource` chapters will appear with their metadata in the future. 

As of now we would be unable to display a scanlator for most sources, because a lot of them don't populate the scanlator field with their name if they are also the scanlators.

Their are also a few aggregator sources that already "incorrectly" populate the scanlator field with their source name.

For use inside `Mihon` the field will only show up in `LocalSource` and even if the source wasn't the original scanlator seeing which source the manga is from looks better / gives the user more information then just leaving it blank. 

We could make it standard policy for sources that are also scanlators to populate the scanlator field with their source name and for sources that aren't to not populate it, if we don't want to risk "wrong" translator names. Whether or not the 3rd party extension will actually follow it is a different question but it can't hurt to try.

We could alternatively create another non schema field and use this for our `LocalSource` when no scanlator is available.

Tell me what you think and I'll make the necessary changes.

**Comment by AntsyLich** [comment]
Could do a custom field then for source and in local source do `scanlator ?: mihon_source`

**Comment by Shamicen** [comment]
> Could do a custom field then for source

I implemented it

File: core-metadata/src/main/java/tachiyomi/core/metadata/comicinfo/ComicInfo.kt [hunk 1]
```diff
@@ -154,6 +156,10 @@ data class ComicInfo(
     @Serializable
     @XmlSerialName("Categories", "http://www.w3.org/2001/XMLSchema", "ty")
     data class CategoriesTachiyomi(@XmlValue(true) val value: String = "")
+
+    @Serializable
+    @XmlSerialName("SourceMihon", "http://www.w3.org/2001/XMLSchema", "ty")
```
Comment (AntsyLich) [inline core-metadata/src/main/java/tachiyomi/core/metadata/comicinfo/ComicInfo.kt:161]:
```suggestion
    @XmlSerialName("SourceMihon", "http://www.w3.org/2001/XMLSchema", "mh")
```

File: app/src/main/java/eu/kanade/domain/manga/model/Manga.kt [hunk 2]
```diff
@@ -95,7 +95,13 @@ fun Manga.hasCustomCover(coverCache: CoverCache = Injekt.get()): Boolean {
 /**
  * Creates a ComicInfo instance based on the manga and chapter metadata.
  */
-fun getComicInfo(manga: Manga, chapter: Chapter, chapterUrl: String, categories: List<String>?) = ComicInfo(
+fun getComicInfo(
+    manga: Manga,
+    chapter: Chapter,
+    urls: String,
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/domain/manga/model/Manga.kt:101]:
```suggestion
    urls: List<String>,
```

File: app/src/main/java/eu/kanade/domain/manga/model/Manga.kt [hunk 3]
```diff
@@ -105,7 +111,7 @@ fun getComicInfo(manga: Manga, chapter: Chapter, chapterUrl: String, categories:
             ComicInfo.Number(it.toString())
         }
     },
-    web = ComicInfo.Web(chapterUrl),
+    web = ComicInfo.Web(urls),
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/domain/manga/model/Manga.kt:114]:
```suggestion
    web = ComicInfo.Web(urls.joinToString(" ")),
```

File: app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt [hunk 4]
```diff
@@ -626,9 +628,22 @@ class Downloader(
...
     ) {
-        val chapterUrl = source.getChapterUrl(chapter.toSChapter())
         val categories = getCategories.await(manga.id).map { it.name.trim() }.takeUnless { it.isEmpty() }
-        val comicInfo = getComicInfo(manga, chapter, chapterUrl, categories)
+        val urls = listOf(source.getChapterUrl(chapter.toSChapter()))
+            .plus(getTracks.await(manga.id).mapNotNull { it.remoteUrl.takeUnless { url -> url.isEmpty() } })
+            .map { it.trim() }
+            .distinct()
+            .joinToString(" ")
+            .trim()
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt:637]:
```suggestion
        val urls = getTracks.await(manga.id)
            .mapNotNull { track ->
                track.remoteUrl.takeUnless { url -> url.isBlank() }?.trim()
             }
            .plus(source.getChapterUrl(chapter.toSChapter()).trim())
            .distinct()
```

**Review by AntsyLich — CHANGES_REQUESTED** [review]

File: app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt [hunk 5]
```diff
@@ -626,9 +628,22 @@ class Downloader(
         chapter: Chapter,
         source: HttpSource,
     ) {
-        val chapterUrl = source.getChapterUrl(chapter.toSChapter())
         val categories = getCategories.await(manga.id).map { it.name.trim() }.takeUnless { it.isEmpty() }
-        val comicInfo = getComicInfo(manga, chapter, chapterUrl, categories)
+        val urls = getTracks.await(manga.id)
+            .mapNotNull { track ->
+                track.remoteUrl.takeUnless { url -> url.isBlank() }?.trim()
+             }
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/Downloader.kt:635]:
```suggestion
            }
```

**Pull request merged**